contracts/stellar-interchain-token-service/src/tests/flow_limit.rs

```diff
@@ -146,6 +149,120 @@ fn execute_its_transfer(
     )
 }
 
+#[derive(Debug, Clone)]
+enum Flow {
+    In(i128),
+    Out(i128),
+}
+
+impl Flow {
+    const fn flip(self) -> Self {
+        match self {
+            Self::In(amount) => Self::Out(amount),
+            Self::Out(amount) => Self::In(amount),
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+struct TestCase {
+    flow_limit: i128,
+    flows: std::vec::Vec<Flow>,
+}
+
+impl TestCase {
+    /// Flips the direction of the flows in the test case. The test case should still logically behave in the same way.
+    fn flip(self) -> Self {
+        Self {
+            flow_limit: self.flow_limit,
+            flows: self.flows.into_iter().map(|flow| flow.flip()).collect(),
+        }
+    }
+}
+
+fn execute_test_case(
+    env: &Env,
+    client: &InterchainTokenServiceClient,
+    gateway: &GatewayConfig,
+    token: &TokenConfig,
+    test_case: TestCase,
+    expected_error: Option<ContractError>,
+) {
+    println!("Executing test case: {:?}", test_case);
+
+    client
+        .mock_all_auths()
+        .set_flow_limit(&token.id, &Some(test_case.flow_limit));
+
+    let (destination_chain, destination_address, data) = dummy_transfer_params(env);
+    let token_address = client.registered_token_address(&token.id);
+
+    if !client.is_trusted_chain(&destination_chain) {
+        client
+            .mock_all_auths()
+            .set_trusted_chain(&destination_chain);
+    }
+
+    #[allow(clippy::manual_try_fold)]
+    let result = test_case
+        .flows
+        .into_iter()
+        .fold(Ok(Ok(())), |previous_result, flow| {
+            assert_ok!(assert_ok!(previous_result));
+
+            println!("Executing flow: {:?}", flow);
+
+            match flow {
+                Flow::In(amount) => {
+                    let msg = approve_its_transfer(env, client, gateway, &token.id, amount);
+
+                    client.try_execute(
+                        &msg.source_chain,
+                        &msg.message_id,
+                        &msg.source_address,
+                        &msg.payload,
+                    )
+                }
+                Flow::Out(amount) => {
+                    let gas_token = setup_gas_token(env, &token.deployer);
+
+                    let token_client = InterchainTokenClient::new(env, &token_address);
+                    token_client.mock_all_auths().mint(&token.deployer, &amount);
+
+                    client.mock_all_auths().try_interchain_transfer(
+                        &token.deployer,
+                        &token.id,
+                        &destination_chain,
+                        &destination_address,
+                        &amount,
+                        &data,
+                        &Some(gas_token),
+                    )
+                }
+            }
+        });
+
+    if let Some(expected_error) = expected_error {
+        assert_contract_err!(result, expected_error)
+    } else {
+        assert_ok!(assert_ok!(result));
+    }
```

Comment:
perhaps using `match` would be simpler?
```suggestion
    match expected_error {
        Some(error) => assert_contract_err!(result, error),
        None => assert_ok!(assert_ok!(result)),
    }
```

Reply:
sure, sometimes clippy suggests collapsing a simple match into an `if let else`, but seems fine in this case